File: GeeksforGeeks_Solutions/python/October_01.py
class Solution:
    def multiply_two_lists(self, first, second):

        first_num,second_num = 0, 0
        
        while first or second:
            first_num = (first_num * 10 + first.data if first else first_num)
            second_num = (second_num * 10 + second.data if second else second_num)
            if first : first = first.next
            if second : second = second.next
            
        return (first_num*second_num) % (10**9 + 7)

File: GeeksforGeeks_Solutions/C++/October_01.cpp
#include <iostream>
using namespace std;

class solution
{
public:
    long long list(Node *node)
    {
        long long sum = 0;
        while (node != NULL)
        {
            sum = sum * 10 + node->data;
            node = node->next;
        }
        return sum;
    }
    long long multiplyTwoLists(Node *first, Node *second)
    {
        long long one = list(first);
        long long two = list(second);
        return (one * two) % 1000000007;
    }
};